vendor/engines/split_accounts/app/models/split_accounts/split_account.rb

```diff
@@ -27,5 +29,16 @@ def extra_penny_count
       splits.select(&:extra_penny?).size
     end
 
+    # Stopped using SQL because that didn't seem to work until the built splits
+    # and subaccounts were created.
+    #
+    # SQL version:
+    #   subaccounts.where("expires_at IS NOT NULL").order(expires_at: :asc).first
+    #
+    def earliest_expiring_subaccount
+      subaccounts = splits.map{ |split| split.subaccount }.select(&:expires_at)
```

Comment:
This confuses me a little. It looks like you'll come out of this as an array of dates. are we not using this method yet?

Can we use the `has_many :subaccounts, through :splits`?
`subaccounts.min_by(&:expires_at)`?